src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -242,14 +244,22 @@ async def _load_scheduler(
         db_conn: SAConnection,
         sgroup_name: str,
     ) -> AbstractScheduler:
-        query = (
-            sa.select([scaling_groups.c.scheduler])
-            .select_from(scaling_groups)
-            .where(scaling_groups.c.name == sgroup_name)
-        )
-        result = await db_conn.execute(query)
-        scheduler_name = result.scalar()
-        return load_scheduler(scheduler_name, self.shared_config['plugins']['scheduler'])
+        try:
+            query = (
+                sa.select([scaling_groups.c.scheduler, scaling_groups.c.scheduler_opts])
+                .select_from(scaling_groups)
+                .where(scaling_groups.c.name == sgroup_name)
+            )
+            result = await db_conn.execute(query)
+            row = result.one()
+            scheduler_name = row['scheduler']
+            scheduler_opts = row['scheduler_opts']
+        except MultipleResultsFound as e:
+            log.exception('_load_scheduler(): multiple scheduler! \n{}', repr(e))
+        except NoResultFound as e:
+            log.exception('_load_scheduler(): scheduler does not exist! \n{}', repr(e))
```

Comment:
It is a good practice to handle potential errors explicitly, but in this case you may assume that there will be always a single row, because the `name` column in the `scaling_groups` table is the primary key and this function is already called from an iteration loop over the list of existing ("schedulable") scaling groups.
(FYI: "schedulable" here means that we consider only the scaling groups which has at least one ALIVE agent)